### PR TITLE
RIIR update lints: Generate modules section and lint group sections

### PR DESCRIPTION
**Pull request by phansch**
This adds the last missing parts of the generating code.

cc #2882

### CODE REVIEW

File: clippy_dev/src/lib.rs [hunk 1]
```diff
@@ -72,6 +72,19 @@ impl Lint {
     }
 }
 
+/// Updates the `pub mod module_name` list in `clippy_lints/src/lib.rs`.
```
Comment (phansch) [inline clippy_dev/src/lib.rs]: s/updates/generates

**Comment by phansch** [comment]
This now also includes the generation of the `register_lint_group` section.

File: clippy_dev/src/main.rs [hunk 2]
```diff
@@ -19,12 +19,17 @@ fn main() {
     let matches = App::new("Clippy developer tooling")
         .subcommand(
             SubCommand::with_name("update_lints")
-                .about("Update the lint list")
+                .about("Makes sure that:\n \
+                       * the lint count in README.md is correct\n \
+                       * the changelog contains markdown link references at the bottom\n \
+                       * all lints groups include the correct lints\n \
```
Comment (flip1995) [inline clippy_dev/src/main.rs]: NIT:
```suggestion
                       * all lint groups include the correct lints\n \
```

r=me with that change

**Comment by phansch** [comment]
bors r=flip1995

**Pull request merged**